python/ray/train/lightning/_lightning_utils.py

```diff
@@ -103,10 +109,20 @@ def distributed_sampler_kwargs(self) -> Dict[str, Any]:
         )
 
     def lightning_module_state_dict(self) -> Dict[str, Any]:
-        """Gathers the full state dict to rank 0 on CPU."""
+        """Gathers the full state dict to rank 0 on CPU.
+
+        Lightning doesn't support FSDP state_dict_type until 2.1.
+        (https://github.com/Lightning-AI/pytorch-lightning/pull/17623).
+        We need a patch logic to enable FSDP checkpointing for Lightning 2.0.x.
```

Comment:
FSDP checkpointing is broken in Lightning 2.0.x. This subclass patches the behavior to perform a full state dict checkpoint, gathering shards on rank 0 CPU. Upgrade to lightning>=2.1 to do sharded state dict checkpointing. See the note in the class docstring for more details.

Reply:
@woshiyyya Can we update this comment to this instead?

Reply:
Updated. It'd be fine since these comments are only for internal developer.